text/0000-export-report.md

```diff
@@ -0,0 +1,82 @@
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The lifecycle exporter/creator binary will accept an optional flag `report` which will provides a path to `report.toml`. When the flag is supplied the export stage will write a toml representation of the built image for platform consumption. When the flag is not provided it will default to writing report.toml in the working directory which by convention is `/layers`.
+
+Here is an overview of the suggested schema for report.toml:
+
+On a build published to a registry:
+```
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+digest = "sha256:c8be1b8f4d60d99c281fc2db75e0f56df42a83ad2f0b091621ce19357e19d853"
+```
+
+On a build exported to the docker daemon:
+```
+[image]
+tags = ["index.docker.io/image/name:latest", "index.docker.io/image/name:other-tag"]
+image_id = "9c0e1895b90f"
+```
+
+By representing this in toml, we can extend this information with additional metadata when needed. A general philosophy for report.toml is that it should contain information that is not accessible elsewhere. For example, the bill of materials should not be included as it is available as an image label and would be unnecessarily duplicated in report.toml. 
+
+# How it Works
+[how-it-works]: #how-it-works
+
+The export stage will write report.toml after exporting and caching is complete. If the export steps fails to complete no report.toml is expected to be written.
+
+This addition will require a bump in the platform api.
+
+Docker daemon based platforms such as `pack` can read the report.toml by utilizing [docker cp](https://docs.docker.com/engine/reference/commandline/cp/).
+
+Kubernetes based platforms can utilize the [pod termination message](https://kubernetes.io/docs/tasks/debug-application-cluster/determine-reason-pod-failure/#customizing-the-termination-message) to retrieve the reports on build completion.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This will require an additional flag which might complicate the lifecycle interface.
+
+# Alternatives
+[alternatives]: #alternatives
+
+- Not doing anything and continuing to require platforms to retrieve this information via alternative means.
+- Writing only the built image identifier to an unstructured file. This would be simpiler for some platforms to parse but, would limit the options for extensibility.
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [The kbld tool parses the pack output with a regex to find the built image digest](https://github.com/k14s/kbld/blob/5597786d8369e966f2e7217b24fd058f4a910675/pkg/kbld/image/pack.go#L17). 
+- Tekton allows tasks to surface a built image digest with an [OCI image layout index.json](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md#surfacing-the-image-digest-built-in-a-task).
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- How/if `pack` should expose this information to users in a programmatic way?
```

Comment:
Do we want to resolve this before accepting the RFC or is that just a separate RFC?

Reply:
As this is primarily requested to enable workflows in kubernetes like platforms that invoke the lifecycle directly, I would like to push that to a separate rfc if possible.

Reply:
@jromero thoughts on inclusion in `pack`?

Reply:
The only case in which we'd be likely to use this would be for something like the [`quiet` option proposed](https://github.com/buildpacks/pack/issues/285) where we'd silence the output of the lifecycle and only use the information provided by this file to output only the desired information. _(This could have also been achieved via structured logging. 😄)_

I would agree with @matthewmcnew that pack is not the primary target for this feature and hence fleshing that out isn't necessary.

Reply:
@matthewmcnew can you put in here that this question is out of scope.

Reply:
Updated to indicate that this is out of scope for this RFC